### ISSUE TITLE
On connect first frame is not sent

### ISSUE THREAD

**Issue by stoefln** [openstf/minicap#153]
**When I connect to the device, it does not instantly send data**
I first have to interact with the screen of the device to get frames to be sent to the client.
Is there a way to force sending data even though the screen didn't change?

**Have you tried STF?**
Sorry, I haven't. Had issues setting up STF and therefore would like to save the time to debug it, since I don't need STF.

**Comment by sorccu** [comment]
If you reconnect to a running minicap you might not get a frame. If you connect for the first time after starting minicap you should always get a frame.

What STF does, though for different reasons, is that it ends the minicap process if the screen is not being viewed. Therefore every time STF connects to minicap it’s connecting to a fresh instance and is guaranteed to receive a frame. You could do something similar, or you could modify the source code to always cache the latest frame, and serve that to new clients.

Basically it’s working as intended, or at least it isn’t a problem for us.

**Comment by stoefln** [comment]
@sorccu one more question relating this topic: How can I end the minicap process? It seem just starting the process again does not help...

**Comment by sorccu** [comment]
Kill it. That’s why the pid is included in the output and header.
